roles/wireguard/templates/iface_config.conf.j2

```diff
@@ -17,28 +17,40 @@ Table = {{ item.table }}
 
 {% for peer in item.peers %}
 [Peer]
-PublicKey = {{ hostvars[peer].wg_pubkeys[item.iface_name] }}
+PublicKey = {{ hostvars[peer].wg_pubkeys[item.iface_name] | default(wireguard_config_directory[peer][item.iface_name].wg_pubkey) }}
 {% if item.psk_enabled is defined %}
-PreSharedKey = {{ hostvars[peer].pre_shared_keys[item.iface_name] }}
+PreSharedKey = {{ hostvars[peer].pre_shared_keys[item.iface_name] | default(wireguard_config_directory[peer][item.iface_name].pre_sharing_key) }}
 {% endif %}
-AllowedIPs = {{ (hostvars[peer].wg_ifaces
-    | selectattr('iface_name', 'equalto', item.iface_name)
-    | map(attribute='ip_addresses_cidr')
-    | first
-    + (item.allowed_ips | default([])))
-    | list
-    | join(', ')
+AllowedIPs = {{
+    (
+      (hostvars[peer].wg_ifaces
+        | selectattr('iface_name', 'equalto', item.iface_name)
+        | map(attribute='ip_addresses_cidr')
+        | first
+      + (item.allowed_ips | default([])))
+      | list
+    ) | default(
+      (wireguard_config_directory[peer][item.iface_name].ip_addresses_cidr
+      + (item.allowed_ips | default([])))
```

Comment:
`item.allowed_ips` will always be empty when this is reached. If it was non-empty, this default is not reached. Therefore, it has no effect to add this here. Probably, you want to factor out the addition of `item.allowed_ips` like the addition of `join` was factored out.

Reply:
I optimized the Jinja2 code accordingly and also made all indents consistent with the rest as well.